crud/src/main/java/com/redhat/lightblue/crud/mongo/Translator.java

```diff
@@ -693,45 +713,148 @@ private String writeJSForLoop(StringBuilder bld, Path p, String varPrefix) {
         return arr.toString();
     }
 
-    private DBObject translateFieldComparison(FieldComparisonExpression expr) {
-        StringBuilder str = new StringBuilder(128);
+    private static final String ARR_ARR_EQ= "if(this.f1.length==this.f2.length) { "+
+        "  var allEq=true;"+
+        "  for(var i=0;i<this.f1.length;i++) { "+
+        "     if(this.f1[i] != this.f2[i]) { allEq=false; break; } "+
+        "  } "+
+        "  if(allEq) return true;"+
+        "}";
+
+    private static final String ARR_ARR_NEQ="if(this.f1.length==this.f2.length) { "+
+        "  var allEq=true;"+
+        "  for(var i=0;i<this.f1.length;i++) { "+
+        "     if(this.f1[i] != this.f2[i]) { allEq=false; break; } "+
+        "  } "+
+        "  if(!allEq) return true;"+
+        "} else { return true; }";
+
+    private static final String ARR_ARR_LT="var n=this.f1.length<this.f2.length?this.f1.length:this.f2.length;"+
+        "var allOk=true;"+
+        "for(var i=0;i<n;i++) {"+
+        "  if(this.f1[i] >= this.f2[i]) { allOk=false;break;}"+
+        "}"+
+        "if(allOk&&(this.%s.length <= this.%s.length)) return true;";
```

Comment:
Fixed that, and started adding some unit test. Re: array length comparison,
i'll look at that

On Wed, Feb 11, 2015 at 1:48 PM, Naveen Malik <notifications@github.com>
wrote:

> In crud/src/main/java/com/redhat/lightblue/crud/mongo/Translator.java
> <https://github.com/lightblue-platform/lightblue-mongo/pull/67#discussion_r24531779>
> :
>
> > +        "}";
> > +
> > +    private static final String ARR_ARR_NEQ="if(this.f1.length==this.f2.length) { "+
> > +        "  var allEq=true;"+
> > +        "  for(var i=0;i<this.f1.length;i++) { "+
> > +        "     if(this.f1[i] != this.f2[i]) { allEq=false; break; } "+
> > +        "  } "+
> > +        "  if(!allEq) return true;"+
> > +        "} else { return true; }";
> > +
> > +    private static final String ARR_ARR_LT="var n=this.f1.length<this.f2.length?this.f1.length:this.f2.length;"+
> > +        "var allOk=true;"+
> > +        "for(var i=0;i<n;i++) {"+
> > +        "  if(this.f1[i] >= this.f2[i]) { allOk=false;break;}"+
> > +        "}"+
> > +        "if(allOk&&(this.%s.length <= this.%s.length)) return true;";
>
> Missed some "%s" in ARR_ARR_LT
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-mongo/pull/67/files#r24531779>
> .
>